Js2 PR

devholdt opened the pull request.

devholdt committed:
Initial commit

devholdt committed:
Initial Commit v2

Added files, installed dependencies and began working on a prototype.

- Added root files for HTML and SCSS
- Installed dependencies for Bootstrap, Sass and Live-Server
- Made sure all needed files are imported for Sass
- Added some Bootstrap and Sass to make a simple layout idea

devholdt committed:
JSDoc commit 1

- Added script to prepare for using JSDoc at a later time
- Created script.js file (with JSDoc lesson task)
- Added script.js CDN to HTML file
- Added conf.json file to make sure node_modules is excluded when creating JSDocs

devholdt committed:
Update README.md

devholdt committed:
Update v0.1

Continued on home page layout

- Added form for posting content (WIP)
- Added form for searching posts or users (WIP)
- Added mockup of how posts will look on the page (version 1)
- Minor changes to styling on nav menu
- Removed unused files

devholdt committed:
Merge branch 'js2' of https://github.com/MHerholdt94/js2-ca into js2

devholdt committed:
Update v0.11 [06.09.2023]

- Added profile.html with a basic layout
- Optimized FontAwesome icon loading, as they loaded a second after everything else. Should work fine now.
- Removed links from nav that probably wont be used
- Added title to a few buttons
- Added an .active class to nav links

devholdt committed:
Update v0.2 - Initial JavaScript [07.09.2023]

Added JavaScript code for rendering menu, posting content to local storage and getting from storage to display on home page

devholdt committed:
Update v0.21 - Initial JavaScript [07.09.2023]

Removed posts.html (intending to have that content in profile.html), added login.html with a simple login form to work with for now.

devholdt committed:
Update v0.22 - Initial JavaScript [07.09.2023]

- Error message when not having anything in one or both login fields
- Bit of styling and layout on profile.html
- Added headings to all pages
-

devholdt committed:
Update README.md

devholdt committed:
Update v0.3 [14.09.2023]

- Finished WIP version of profile page
- Compressed placeholder profile pic
- Added visible profile info above nav menu
- Added profile pic to post cards
- Decided against having usernames, but instead use email address as usernames (for now)
- Minor changes to styling and Sass files to be cleaner

devholdt committed:
Merge branch 'js2' of https://github.com/MHerholdt94/js2-ca into js2

devholdt committed:
Update v.0.31 [14.09.2023]

- Added signup.html file with a signup form (WIP)
- Redone form validation for both login form and signup form
- Added formValidation.js file for form validation function
- Added "Sign up" link in nav, goal is to make nav dynamic depending on if a user is logged in or not

devholdt committed:
User registration and login

Added JS for user registration and login with calls to the provided API.

Minor changes to variable names, placeholder texts, HTML element IDs and some

devholdt committed:
Display posts, local storage and updated nav

Posts from the API is now displayed on the home page (the last 10 for now)

When logged in, user info (object) and token is saved in local storage for easy fetchin'

Nav will now show the profile pic and username of user if they're logged in (nothing if not)

Better error messages for posts display and login (in case of API call errors)

devholdt committed:
Updated nav, local storage

More dynamic nav, showing certain links depending on if a user is logged in or not. With better renderMenu code

Updated storage.js to fit better with the current user login/registration/interaction

Profile with more correct profile info is shown on all pages if logged in (nothing if not)

devholdt committed:
Updated login + flexible HTTP requests

Updated login function to show clearer messages depending on:
- no values to inputs
- wrong login credentials
- successful login

If login was successful, displays a welcome message with username. After 3 seconds location change to home page.

HTTP requests are made dynamic and flexible with its own exported function in a separate file "httpRequest.js".

Added JSDoc to httpRequest function.

Removed formValidation.js and its imports across some files. Also updated forms in HTML to accomodate this.

Made a separate file for URLs and other constants in the settings folder (constants.js)

Fixed navProfile element to only display if a user is logged in

Removed some unused code

devholdt committed:
Updated user registration

The user registration function now also uses the flexible httpRequest function to register users.

It also displays correct messages depending on if there are no values added to the inputs, the values are invalid or if they are valid and a user is registered.

If a user registration was successful, the user is logged in automatically and taken to the home page.

devholdt committed:
Updated posts display

renderPosts function in index.js is better now. Also added a message where the posts should be if user isn't logged in telling them to log in to see posts.

Removed unused code and asset

devholdt committed:
Updated profile page and cleanup

Profile page does now actually include the username, email, avatar and banner of the user with a neater layout.

Added a default banner image if user havent added it yet (WIP).

Users will be redirected to other pages if they try to access i.e. profile.html without being logged in.

nav-profile is now user-info and it's moved to the header (top left of pages).

General cleanup in the .scss department

devholdt committed:
Avatar and banner update pt1

Added ability to add avatar and banner image URLs during user registration. The inputs should have correct validation, so they will only accept publicly available URLs to images. I also added a button for each input to quickly clear the inputs, since URLs tend to be long. These inputs are optional and not required to finish user registration.

If a user adds URLs for avatar and/or banner, they will correctly show up as either their avatar across all pages and/or their profile banner on the profile page (depending on if they added URLs for only one of them or both).

Updated "message" function to only appear in ".message-container", since the plan is to always use that element. Also added a timer so when a message appears it dissapears after 3 seconds.

Added text and link to registration form and login form asking the user if they already have an account or need to create one, with links.

Other style changes across all pages

devholdt committed:
Create .gitignore

devholdt committed:
Test login update

Made a testlogin.html file, will try to make a better login/registration page

devholdt committed:
UrSpace v2

Redesigned the site to only have 3 pages, index, feed and profile (might be more later).

index.html is now the login and registration page. Users will have to log in or register a user to enter the site. This page has a new and cleaner layout with possibilities of both logging in and registering a new user.

feed.html is the new file name of the old index.html. No big changes to this page and profile.html.

Added new .svg logo for all pages with a slightly newer design.

Minor changes to file structure as well as removal of these now unused files:
- login.html
- signup.html
- login.js
- signup.js
- testlogin.html (only used when creating new layout)

devholdt committed:
Post content update

Logged in users can now post content to the API. Posts include a title and optional body text and media (image).

Updated message style to look cleaner and more on-brand.

Added some more color variables for the messages.

Other minor changes like:
- Removed unused files
- Minor changes to styles (messages, cards)
- Message display time
- More...

devholdt committed:
cleaning duty

Minor style change to posts (cards)
Removed unused file

devholdt committed:
Posts feed pagination

Added pagination buttons for the posts feed. Prev and Next buttons as well as pages 1 through 10 with 10 posts on each page.

devholdt committed:
Post media update

Better media display on posts, with just a thumbnail version of the media being shown on each post. Users can click the thumbnail to open a modal of the full sized image.

devholdt committed:
Updated profile, better post feed

Added users own posts displayed on their profile.

Better pagination and post feed in general, minor changes to style on post cards (added buttons again).

devholdt committed:
Added delete function

Added function to delete users own posts from the profile page (WIP)

devholdt committed:
Better feed, post form, cards and delete function

New layout for the feed page
New styling for post cards which now also includes:
- Poster name
- Date and time it was posted
- Neater buttons for follow, edit and comment
- Dynamic post cards depending on if the post is the logged in users posts or not
- User is now asked to confirm the deletion of a post before its deleted
- Made center column for feed.html and profile.html wider
- Added a "rules and requirements" column next to the post form on feed page

devholdt committed:
Form post small update

Minor changes to form post style

devholdt committed:
Broken follow/unfollow update

Added some VERY work-in-progress follow/unfollow system that, as of right now, is pretty broken but also works a little bit. But not really.

devholdt committed:
Edit posts update

Added possibility of editing logged in users own posts. Still cant close the edit form modal yet, but one day you will

devholdt committed:
Tags part one update

Added possibility to add a tag to posts, as well as edit posts to add/remove tag.

Also fixed the edit modal to be closeable

Made a new API_URLS object in constans.js to make importing API URLs a bit easier.

devholdt committed:
Search update part 1

Added semi-working search function. Users can now search by:
- Username
- Post title
- Post tags

...using the search input in feed.html.

Currently if it returns more than 10 results, the user can click a "more results" button that shows the next 10 results. The goal is to have these 10 results appear below the first 10 results instead of replacing them as they do now.

Also updated the exported message function to have more flexible setTimout setting to them.

Minor changes to style, scss variables, post tags etc. as well as the post fort on feed.html

devholdt committed:
Search function update 1

Updated search functionality [TBC]

devholdt committed:
Oh yeah search babyyyyy

Search function woooooo :)

devholdt committed:
Search v2.0 and more

Moved search functionality away from feed.html and into a new html page for searching/filtering as issues became too complex to fix with pagination.

Added a "scroll to top" button for searchResults.html and feed.html that displays at set scrollY positions and scrolls the user to a set scroll target.

devholdt committed:
Small position commit

Changes to position of some elements.
- Removed sticky position from top pagination
- Added sticky position to nav on all pages

devholdt committed:
Search done and new fetch/render post functions

Search functionality should work as intended now.

Made a separate function/file for fetching posts from the URL and one for rendering the posts as it got repeated quite a lot. Also made displaying image modals for posts its own file and exported function.

Renamed generatePostsUrl to paginationUrl as the name suggested it was in general when it was really only for pagination.

Removed unused handleSearch.js file

Other minor cleanups

devholdt committed:
gitignore commit

devholdt committed:
New post display update

Branch with a new kind of post display without the old pagination. Might be easier regarding search/filter function without the pagination.

Added possibility to search/filter post ID as well as title, username and tags as before.

New file renderPosts.js now has the functions for displaying posts as well as a separate function for rendering the post content itself.

Removed now unused files pagination.js and fetchPosts.js.

Added JSDoc and comments to a few functions. Goal is to have JSDoc on most of or all functions and more comments for description.

Minor changes to style etc and cleanup.

devholdt committed:
Change images, JSDocs/comments

Added possibility of changing avatar image and banner image from a "settings" button on the profile page. The form inputs accepts the same kind of image URLs as the login, registration and post forms do.

Also added JSDocs to most/all functions as well as more comments throughout the code to hopefully make it easier to understand.

devholdt committed:
More profile content

Added more details to the profile page

devholdt committed:
Comments commit

Added comment sections to posts as well as a comment input to add comments to posts.

Removed all console.logs that were deemed unnecessary and instead updated all message functions to also include the actual error message.

devholdt committed:
Post reactions commit pt.1

Added possibility of post reactions (likes). Not fully developed, but should work as intended other than minor inconsitensies.

Minor changes to post content layout to accomodate these changes.

devholdt committed:
Merge pull request #1 from MHerholdt94/new-post-display

New post display

devholdt committed:
Update README.md

devholdt committed:
Update README.md

devholdt committed:
Dynamic profiles

Profile page will now display profile data of the user profile you are visiting. If user clicks on the "profile" link in the nav or their own username in the header they will visit their own profile page.

devholdt committed:
Remove node_modules from version control

devholdt committed:
gitignore

devholdt committed:
Remove node_modules from version control

devholdt committed:
gitignore

Fermain commented on feed.html:
```diff
@@ -0,0 +1,144 @@
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>UrSpace | Home</title>
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.3/dist/css/bootstrap.min.css" rel="stylesheet"
```
You shouldn't need this since bootstrap is in your dist file already
```suggestion
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.3/dist/css/bootstrap.min.css" rel="stylesheet"
```

Fermain commented on feed.html:
```diff
@@ -0,0 +1,144 @@
+                <div class="feed-left-column position-sticky">
+                    <nav class="nav"></nav>
+
+                    <!-- <div class="sort mt-5">
```
If you aren't going to use it, put it in another branch and remove it from submission

Fermain commented on index.html:
```diff
@@ -0,0 +1,106 @@
+            <p class="message-container message-index"></p>
+        </div>
+
+        <form id="loginForm">
```
```suggestion
        <form id="loginForm" action="profile.html">
```

Try this

Fermain commented on searchResults.html:
```diff
@@ -0,0 +1,86 @@
+
+            </div>
+
+            <div class="col"></div>
```
?

devholdt replied:
The main tags is divided into three columns, so line 67 is the 3rd column but it has no content on the this page and profile page. Removing it causes the 2nd column to not be centered/1st column to take up too much space.
I suppose there are better ways to do this but I haven't looked for it yet.

Fermain commented on searchResults.html:
```diff
@@ -0,0 +1,86 @@
+
+            <div class="col"></div>
+
+        </div>
```
Indentation - Please run a formatter on this file again

MHerholdt94 replied:
The <main> is divided into three columns, so line 67 is the 3rd column but it has no content on the this page and profile page. Removing it causes the 2nd column to not be centered.
I suppose there are better ways to do this but I haven't looked for it yet.

Fermain reviewed (commented):
Partial review, looking through the JS now

devholdt committed:
Follow/Unfollow fixed

Following/Unfollowing profiles is now working as intended and can be done by visiting a users profile page and click the follow/unfollow button.

Users followed and users following you appear on the profile page with links to each of their profiles.

Removed unused code

devholdt committed:
Follow/Unfollow update v2

Follow/unfollow elements are no longer sticky, as it made problems if user followed/were followed by a lot of users. Instead they now have a max height and they are scrollable if it goes above that max height.

Made default image urls an exported variable in the constants.js file.

devholdt committed:
Merge pull request #3 from MHerholdt94/new-post-display

New post display

devholdt committed:
Update index.html

Co-authored-by: Oliver Dipple <Fermain@users.noreply.github.com>

devholdt committed:
Update README.md

devholdt committed:
Hamburger menu woop

The navigation menu now appear as a hamburger menu instead on smaller screen sizes (998px and lower). User info (in the header) also appears inside that menu instead.

Fixed "Profile" link not displaying active state properly.

Changed width of modals to be more responsive.

Removed unused code and image.

devholdt committed:
Search and filter finished

Search/filter should now work a lot better and will display both posts and users that fit the search query

devholdt committed:
Merge pull request #4 from MHerholdt94/new-post-display

Hamburger menu woop

devholdt committed:
Responsive commit

Made site more responsive overall (down to 360px width)

devholdt committed:
dynamic title for profile page

HTML title now changes depending on which profile user is currently visiting

devholdt committed:
Merge pull request #5 from MHerholdt94/new-post-display

Responsive commit

devholdt committed:
Neater post form

The post form is now "inside" an accordion.
Form requirements are displayed as small info icons the user can hover over to reveal input requirement on larger screens while on smallers screens theres the original requirements list, although beneath the form and not next to it.

Also added a post form to the profile page, however only on the logged-in users page

devholdt committed:
Merge pull request #6 from MHerholdt94/new-post-display

Neater post form

devholdt committed:
Logo link fix

Fixed so clicking the logo (except on index.html) takes user to feed.html instead of index.html

devholdt committed:
Merge pull request #7 from MHerholdt94/new-post-display

Logo link fix

devholdt committed:
Better image validation, fetching and displaying

Rewritten url validation function and made the displaying of profile images (avatars and banners) to fetch from API instead of localstorage

devholdt committed:
Merge pull request #8 from MHerholdt94/new-post-display

Better image validation, fetching and displaying

devholdt committed:
Filter/sort and better message system

Removed searchResults.html/.js/.scss alltogether and instead added the search bar and functionality to the feed.html page.

Added a select element users can use to pick between two options, "all posts" and "following" to sort posts if they want to see all or just from users they follow.

Added a .message-fixed element to all pages that works as a fixed message that is centered on the screen. Messages began to be a bit cluttered with too many message containers.

devholdt committed:
action profile.html

devholdt committed:
Update README.md

devholdt committed:
profile column resize

devholdt committed:
Merge pull request #9 from MHerholdt94/new-post-display

Filter/sort and better message system

devholdt committed:
Error handling, JSDoc and cleanup

Minor changes to error handling to be more user friendly (less technical dill dall).

Added JSDoc and comments to more functions.

General cleanup of code.

devholdt committed:
js to mjs

module js files now have file extension .mjs

devholdt committed:
Update README.md

devholdt committed:
Minor message fix

devholdt committed:
Merge branch 'js2' of https://github.com/MHerholdt94/js2-ca into js2

devholdt committed:
Merge branch 'main' into js2
